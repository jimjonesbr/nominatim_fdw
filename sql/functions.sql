CREATE SERVER osm 
FOREIGN DATA WRAPPER nominatim_fdw 
OPTIONS (url 'https://nominatim.openstreetmap.org');

SELECT * FROM nominatim_fdw_version();

SELECT 
    osm_id, osm_type, ref, class, display_name, display_rank, place_id IS NULL, place_rank,
    lon, lat, boundingbox, importance, icon, timestamp IS NOT NULL AS ts, attribution,
    querystring, length(polygon) AS polygon_length, exclude_place_ids, more_url,
    jsonb_pretty(extratags) AS extratags, jsonb_pretty(namedetails) AS namedetails,
    jsonb_pretty(addressdetails) AS addressdetails
FROM nominatim_search(
      server_name => 'osm',
      q => 'einsteinstraße 60, münster, germany',
      extratags => true,
      addressdetails => true,
      namedetails => true,
      polygon => 'polygon_text',
      email => 'jim.jones@uni-muenster.de',
      countrycodes => 'DE,BR,US',
      featuretype => 'office',
      dedupe => true,
      exclude_place_ids => '42,73',
      viewbox => '51.9659397,51.9661584,7.6036345,7.6039893',
      polygon_threshold => 0.1,
      layer => 'address',
      limit_result => 1,
      accept_language => 'de_DE,de,q=0.9');

SELECT pg_sleep(2);

SELECT 
    osm_id, osm_type, ref, class, display_name, display_rank, place_id IS NULL, place_rank,
    lon, lat, boundingbox, importance, icon, timestamp IS NOT NULL AS ts, attribution,
    querystring, length(polygon) AS polygon_length, exclude_place_ids, more_url,
    jsonb_pretty(extratags) AS extratags, jsonb_pretty(namedetails) AS namedetails,
    jsonb_pretty(addressdetails) AS addressdetails
FROM nominatim_search(
      server_name => 'osm',
      amenity => 'wwu it',
      polygon => 'polygon_text',
      extratags => true,
      addressdetails => true,
      namedetails => true,      
      street => 'einsteinstraße 60',
      city => 'münster',
      state => 'nordrhein westfalen',
      country => 'germany',
      postalcode => '48149',
      email => 'jim.jones@uni-muenster.de',
      countrycodes => 'DE,BR,US',
      featuretype => 'office',
      dedupe => true,
      exclude_place_ids => '42, 73',
      viewbox => '51.9659397,51.9661584,7.6036345,7.6039893',
      polygon_threshold => 0.1,
      layer => 'address',
      limit_result => 1,
      accept_language => 'de_DE,de,q=0.9');

SELECT pg_sleep(2);

SELECT 
    osm_id, osm_type, result, ref, place_id IS NULL, place_rank, lon, lat, boundingbox, 
    icon, timestamp IS NOT NULL AS ts, attribution, querystring, 
    length(polygon) AS polygon_length, jsonb_pretty(extratags) AS extratags, 
    jsonb_pretty(namedetails) AS namedetails, jsonb_pretty(addressparts) AS addressparts
FROM nominatim_reverse(
        server_name => 'osm', 
        lon => 7.6038115,
        lat => 51.9660873,        
        polygon => 'polygon_text',
        extratags => true,
        addressdetails => true,
        namedetails => true,
        accept_language => 'de_DE,de,q=0.9');

SELECT pg_sleep(2);

SELECT 
    osm_id, osm_type, ref, class, display_name, display_rank, place_id IS NULL, place_rank,
    lon, lat, boundingbox, importance, icon, timestamp IS NOT NULL AS ts, attribution,
    querystring, length(polygon) AS polygon_length, exclude_place_ids, more_url,
    jsonb_pretty(extratags) AS extratags, jsonb_pretty(namedetails) AS namedetails,
    jsonb_pretty(addressdetails) AS addressdetails 
FROM nominatim_lookup(
      server_name => 'osm',
      osm_ids => 'W88291927',
      extratags => true,
      addressdetails => true,
      namedetails => true,
      polygon => 'polygon_text',
      email => 'jim.jones@uni-muenster.de',
      countrycodes => 'DE,BR,US',
      featuretype => 'office',
      dedupe => true,
      exclude_place_ids => '42,73',
      viewbox => '51.9659397,51.9661584,7.6036345,7.6039893',
      polygon_threshold => 0.1,
      layer => 'address',
      accept_language => 'de_DE,de,q=0.9');