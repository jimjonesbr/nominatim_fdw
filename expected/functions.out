CREATE SERVER osm 
FOREIGN DATA WRAPPER nominatim_fdw 
OPTIONS (url 'https://nominatim.openstreetmap.org');
SELECT nominatim_fdw_version() IS NOT NULL, 
       nominatim_fdw_version() <> '';
 ?column? | ?column? 
----------+----------
 t        | t
(1 row)

SELECT 
    osm_id, osm_type, 
	ref, 
	class, 
	display_name IS NOT NULL AND display_name <> '' valid_display_name,
	display_rank, 
	place_id IS NOT NULL AND place_id > 0 AS valid_place_id, 
	place_rank,
    lon, lat, boundingbox, 
	importance, 
	icon, 
	timestamp IS NOT NULL AS valid_timestamp, 
	attribution,
    querystring, 
	length(polygon) AS polygon_length, 
	exclude_place_ids IS NOT NULL AS valid_exclude_place_ids, 
	more_url IS NOT NULL AND more_url <> '' AS valid_more_url,
    jsonb_pretty(extratags) AS extratags, 
	jsonb_pretty(namedetails) AS namedetails,
    jsonb_pretty(addressdetails) AS addressdetails
FROM nominatim_search(
      server_name => 'osm',
      q => 'einsteinstraße 60, münster, germany',
      extratags => true,
      addressdetails => true,
      namedetails => true,
      polygon => 'polygon_text',
      email => 'jim.jones@uni-muenster.de',
      countrycodes => 'DE,BR,US',
      featuretype => 'poi',
      dedupe => true,
      exclude_place_ids => '42,73',
      viewbox => '51.9659397,51.9661584,7.6036345,7.6039893',
      polygon_threshold => 0.1,
      layer => 'address',
      limit_result => 1,
      bounded => false,
      accept_language => 'de_DE,de,q=0.9');
  osm_id  | osm_type |  ref   | class  | valid_display_name | display_rank | valid_place_id | place_rank |    lon    |    lat     |                boundingbox                |      importance      | icon | valid_timestamp |                              attribution                              |             querystring             | polygon_length | valid_exclude_place_ids | valid_more_url |                     extratags                     |                      namedetails                       |            addressdetails             
----------+----------+--------+--------+--------------------+--------------+----------------+------------+-----------+------------+-------------------------------------------+----------------------+------+-----------------+-----------------------------------------------------------------------+-------------------------------------+----------------+-------------------------+----------------+---------------------------------------------------+--------------------------------------------------------+---------------------------------------
 88291927 | way      | WWU IT | office | t                  |              | t              |         30 | 7.6038115 | 51.9660873 | 51.9659397,51.9661584,7.6036345,7.6039893 | 9.99999999995449e-06 |      | t               | Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright | einsteinstraße 60, münster, germany |            115 | t                       | t              | {                                                +| {                                                     +| {                                    +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |     "website": "https://www.uni-muenster.de/IT/",+|     "name": "WWU IT",                                 +|     "city": "Münster",               +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |     "building": "university",                    +|     "name:de": "WWU IT",                              +|     "road": "Einsteinstraße",        +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |     "roof:shape": "flat",                        +|     "old_name": "Zentrum für Informationsverarbeitung"+|     "state": "Nordrhein-Westfalen",  +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |     "wheelchair": "yes",                         +| }                                                      |     "office": "WWU IT",              +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |     "roof:colour": "#A9A9A9",                    +|                                                        |     "suburb": "Innenstadtring",      +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |     "old_short_name": "ZIV",                     +|                                                        |     "country": "Deutschland",        +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |     "building:colour": "#D3D3D3",                +|                                                        |     "postcode": "48149",             +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |     "building:levels": "2"                       +|                                                        |     "country_code": "de",            +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                | }                                                 |                                                        |     "house_number": "60",            +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |                                                   |                                                        |     "city_district": "Münster-Mitte",+
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |                                                   |                                                        |     "neighbourhood": "Neutor",       +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |                                                   |                                                        |     "ISO3166-2-lvl4": "DE-NW"        +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                     |                |                         |                |                                                   |                                                        | }
(1 row)

SELECT pg_sleep(2);
 pg_sleep 
----------
 
(1 row)

SELECT 
    osm_id, osm_type, 
	ref, 
	class, 
	display_name IS NOT NULL AND display_name <> '' valid_display_name,
	display_rank, 
	place_id IS NOT NULL AND place_id > 0 AS valid_place_id, 
	place_rank,
    lon, lat, boundingbox, 
	importance, 
	icon, 
	timestamp IS NOT NULL AS valid_timestamp, 
	attribution,
    querystring, 
	length(polygon) AS polygon_length, 
	exclude_place_ids IS NOT NULL AS valid_exclude_place_ids, 
	more_url IS NOT NULL AND more_url <> '' AS valid_more_url,
    jsonb_pretty(extratags) AS extratags, 
	jsonb_pretty(namedetails) AS namedetails,
    jsonb_pretty(addressdetails) AS addressdetails
FROM nominatim_search(
      server_name => 'osm',
      amenity => 'wwu it',
      polygon => 'polygon_text',
      extratags => true,
      addressdetails => true,
      namedetails => true,      
      street => 'einsteinstraße 60',
      city => 'münster',
      state => 'nordrhein westfalen',
      country => 'germany',
      postalcode => '48149',
      email => 'jim.jones@uni-muenster.de',
      countrycodes => 'DE,BR,US',
      featuretype => 'office',
      dedupe => true,
      exclude_place_ids => '42, 73',
      viewbox => '51.9659397,51.9661584,7.6036345,7.6039893',
      polygon_threshold => 0.1,
      layer => 'address',
      limit_result => 1,
      accept_language => 'de_DE,de,q=0.9');
  osm_id  | osm_type |  ref   | class  | valid_display_name | display_rank | valid_place_id | place_rank |    lon    |    lat     |                boundingbox                |      importance      | icon | valid_timestamp |                              attribution                              |                               querystring                               | polygon_length | valid_exclude_place_ids | valid_more_url |                     extratags                     |                      namedetails                       |            addressdetails             
----------+----------+--------+--------+--------------------+--------------+----------------+------------+-----------+------------+-------------------------------------------+----------------------+------+-----------------+-----------------------------------------------------------------------+-------------------------------------------------------------------------+----------------+-------------------------+----------------+---------------------------------------------------+--------------------------------------------------------+---------------------------------------
 88291927 | way      | WWU IT | office | t                  |              | t              |         30 | 7.6038115 | 51.9660873 | 51.9659397,51.9661584,7.6036345,7.6039893 | 9.99999999995449e-06 |      | t               | Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright | wwu it, einsteinstraße 60, münster, nordrhein westfalen, 48149, germany |            115 | t                       | t              | {                                                +| {                                                     +| {                                    +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |     "website": "https://www.uni-muenster.de/IT/",+|     "name": "WWU IT",                                 +|     "city": "Münster",               +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |     "building": "university",                    +|     "name:de": "WWU IT",                              +|     "road": "Einsteinstraße",        +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |     "roof:shape": "flat",                        +|     "old_name": "Zentrum für Informationsverarbeitung"+|     "state": "Nordrhein-Westfalen",  +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |     "wheelchair": "yes",                         +| }                                                      |     "office": "WWU IT",              +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |     "roof:colour": "#A9A9A9",                    +|                                                        |     "suburb": "Innenstadtring",      +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |     "old_short_name": "ZIV",                     +|                                                        |     "country": "Deutschland",        +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |     "building:colour": "#D3D3D3",                +|                                                        |     "postcode": "48149",             +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |     "building:levels": "2"                       +|                                                        |     "country_code": "de",            +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                | }                                                 |                                                        |     "house_number": "60",            +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |                                                   |                                                        |     "city_district": "Münster-Mitte",+
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |                                                   |                                                        |     "neighbourhood": "Neutor",       +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |                                                   |                                                        |     "ISO3166-2-lvl4": "DE-NW"        +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |                                                                         |                |                         |                |                                                   |                                                        | }
(1 row)

SELECT pg_sleep(2);
 pg_sleep 
----------
 
(1 row)

SELECT 
    osm_id > 0 AS valid_osm_id, 
	osm_type, 
	result IS NOT NULL AND result <> '' valid_result,
	ref IS NOT NULL AND ref <> '' valid_ref,
	place_id IS NOT NULL AND place_id > 0 AS valid_place_id, 
	place_rank, 
	lon > 7.6 AS valid_lon, lat > 51.9 AS valid_lat, 
	array_length(string_to_array(boundingbox,','),1) = 4 AS valid_bbox, 
    icon, 
	timestamp IS NOT NULL AS valid_timestamp, 
	attribution IS NOT NULL AND attribution <> '' AS valid_attribution,  
	querystring, 
    length(polygon)>0 AS valid_polygon, 
	jsonb_pretty(extratags) IS NOT NULL AS valid_extratags, 
    jsonb_pretty(namedetails) IS NOT NULL AS valid_namedetails, 
	jsonb_pretty(addressparts) IS NOT NULL AS valid_addressparts
FROM nominatim_reverse(
        server_name => 'osm', 
        lon => 7.6038115,
        lat => 51.9660873,        
        polygon => 'polygon_text',
        extratags => true,
        addressdetails => true,
        namedetails => true,
        accept_language => 'de_DE,de,q=0.9');
 valid_osm_id | osm_type | valid_result | valid_ref | valid_place_id | place_rank | valid_lon | valid_lat | valid_bbox | icon | valid_timestamp | valid_attribution |              querystring              | valid_polygon | valid_extratags | valid_namedetails | valid_addressparts 
--------------+----------+--------------+-----------+----------------+------------+-----------+-----------+------------+------+-----------------+-------------------+---------------------------------------+---------------+-----------------+-------------------+--------------------
 t            | way      | t            | t         | t              |         30 | t         | t         | t          |      | t               | t                 | lat=51.966087&lon=7.603811&format=xml | t             | t               | t                 | t
(1 row)

SELECT pg_sleep(2);
 pg_sleep 
----------
 
(1 row)

		
SELECT 
    osm_id, osm_type, 
	ref, 
	class, 
	display_name IS NOT NULL AND display_name <> '' valid_display_name,
	display_rank, 
	place_id IS NOT NULL AND place_id > 0 AS valid_place_id, 
	place_rank,
    lon, lat, boundingbox, 
	importance, 
	icon, 
	timestamp IS NOT NULL AS valid_timestamp, 
	attribution,
    querystring, 
	length(polygon) AS polygon_length, 
    jsonb_pretty(extratags) AS extratags, 
	jsonb_pretty(namedetails) AS namedetails,
    jsonb_pretty(addressdetails) AS addressdetails
FROM nominatim_lookup(
      server_name => 'osm',
      osm_ids => 'W88291927',
      extratags => true,
      addressdetails => true,
      namedetails => true,
      polygon => 'polygon_text',
      email => 'jim.jones@uni-muenster.de',
      countrycodes => 'DE,BR,US',
      featuretype => 'office',
      dedupe => true,
      exclude_place_ids => '42,73',
      viewbox => '51.9659397,51.9661584,7.6036345,7.6039893',
      polygon_threshold => 0.1,
      layer => 'address',
      accept_language => 'de_DE,de,q=0.9');		
  osm_id  | osm_type |  ref   | class  | valid_display_name | display_rank | valid_place_id | place_rank |    lon    |    lat     |                boundingbox                |      importance      | icon | valid_timestamp |                              attribution                              | querystring | polygon_length |                     extratags                     |                      namedetails                       |            addressdetails             
----------+----------+--------+--------+--------------------+--------------+----------------+------------+-----------+------------+-------------------------------------------+----------------------+------+-----------------+-----------------------------------------------------------------------+-------------+----------------+---------------------------------------------------+--------------------------------------------------------+---------------------------------------
 88291927 | way      | WWU IT | office | t                  |              | t              |         30 | 7.6038115 | 51.9660873 | 51.9659397,51.9661584,7.6036345,7.6039893 | 9.99999999995449e-06 |      | t               | Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright |             |            115 | {                                                +| {                                                     +| {                                    +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |     "website": "https://www.uni-muenster.de/IT/",+|     "name": "WWU IT",                                 +|     "city": "Münster",               +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |     "building": "university",                    +|     "name:de": "WWU IT",                              +|     "road": "Einsteinstraße",        +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |     "roof:shape": "flat",                        +|     "old_name": "Zentrum für Informationsverarbeitung"+|     "state": "Nordrhein-Westfalen",  +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |     "wheelchair": "yes",                         +| }                                                      |     "office": "WWU IT",              +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |     "roof:colour": "#A9A9A9",                    +|                                                        |     "suburb": "Innenstadtring",      +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |     "old_short_name": "ZIV",                     +|                                                        |     "country": "Deutschland",        +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |     "building:colour": "#D3D3D3",                +|                                                        |     "postcode": "48149",             +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |     "building:levels": "2"                       +|                                                        |     "country_code": "de",            +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                | }                                                 |                                                        |     "house_number": "60",            +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |                                                   |                                                        |     "city_district": "Münster-Mitte",+
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |                                                   |                                                        |     "neighbourhood": "Neutor",       +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |                                                   |                                                        |     "ISO3166-2-lvl4": "DE-NW"        +
          |          |        |        |                    |              |                |            |           |            |                                           |                      |      |                 |                                                                       |             |                |                                                   |                                                        | }
(1 row)

